File: src/main/java/io/github/suitetecsa/sdk/nauta/network/HttpResponse.java
package io.github.suitetecsa.sdk.nauta.network;

import org.jetbrains.annotations.Contract;
import org.jetbrains.annotations.NotNull;

import java.nio.charset.StandardCharsets;
import java.util.Arrays;
import java.util.Map;

public record HttpResponse(int statusCode, String statusMessage, byte[] content, Map<String, String> cookies) {
    public HttpResponse(int statusCode, String statusMessage, byte[] content, Map<String, String> cookies) {
        this.statusCode = statusCode;
        this.statusMessage = statusMessage;
        this.content = Arrays.copyOf(content, content.length);
        this.cookies = cookies;
    }

    @Override
    @Contract(value = " -> new", pure = true)
    public byte @NotNull [] content() {
        return Arrays.copyOf(content, content.length);
    }

    @Contract(value = " -> new", pure = true)
    public @NotNull String getText() {
        return new String(content, StandardCharsets.UTF_8);
    }
}

File: src/main/kotlin/io/github/suitetecsa/sdk/nauta/network/HttpResponse.kt
package io.github.suitetecsa.sdk.nauta.network

import org.jetbrains.annotations.Contract
import java.nio.charset.StandardCharsets

class HttpResponse(
    val statusCode: Int,
    val statusMessage: String,
    content: ByteArray,
    val cookies: Map<String, String>
) {
    @Contract(value = " -> new", pure = true)
    fun content(): ByteArray {
        return content.copyOf(content.size)
    }

    @get:Contract(value = " -> new", pure = true)
    val text: String
        get() = String(content, StandardCharsets.UTF_8)
    val content: ByteArray = content.copyOf(content.size)
}